pywr/parameters/_parameters.pyx

```diff
@@ -261,6 +262,111 @@ cdef class ArrayIndexedScenarioParameter(Parameter):
         return self.values[ts._index, scenario_index._indices[self._scenario_index]]
 
 
+cdef class TablesArrayParameter(IndexParameter):
+    def __init__(self, h5file, node, where='/', scenario=None, **kwargs):
+        """
+        This Parameter reads array data from a PyTables HDF database.
+
+        The parameter reads data using the PyTables array interface and therefore
+        does not require loading the entire dataset in to memory. This is useful
+        for large model runs.
+
+        Parameters
+        ----------
+        h5file : tables.File or filename
+            The tables file handle or filename to attach the CArray objects to. If a
+            filename is given the object will open and close the file handles.
+        node : string
+            Name of the node in the tables database to read data from
+        where : string
+            Path to read the node from.
+        scenario : Scenario
+            Scenario to use as the second index in the array.
+        """
+        super(TablesArrayParameter, self).__init__(**kwargs)
+
+        self.h5file = h5file
+        self.h5store = None
+        self.node = node
+        self.where = where
+        self.scenario = scenario
+
+        # Private attributes, initialised during reset()
+        self._values_dbl = None
+        self._values_int = None
+        self._scenario_index = -1
+
+    cpdef setup(self, model):
+        self.model = model
+        self._scenario_index = -1
+        # This setup must find out the index of self._scenario in the model
+        # so that it can return the correct value in value()
+        if self.scenario is not None:
+            self._scenario_index = model.scenarios.get_scenario_index(self.scenario)
+
+    cpdef reset(self):
+        self.h5store = H5Store(self.h5file, None, "r")
+        node = self.h5store.file.get_node(self.where, self.node)
+
+        # detect data type and read into memoryview
+        if node.dtype == np.float32 or node.dtype == np.float64:
+            self._values_dbl = node.read()
+            self._values_int = None
+            shape = self._values_dbl.shape
+        else:
```

Comment:
Should we be explicit about which types to use as ints here? And throw an error it's something else?